File: python/novacrypt.py
class NovaCrypt:
    def __init__(self, SECURE_KEY: int, SALT: int) -> None:
        self.secureKey = SECURE_KEY
        self.salt = 9 if SALT == 0 else SALT

    def encrypt(self, content: str) -> str:
        encrypted_content = ''
        for c in content:
            genSpecCode = (ord(c) + (self.secureKey * self.salt) + self.salt) % 256
            encrypted_content += chr(genSpecCode)
        return encrypted_content

    def decrypt(self, encrypted_content: str) -> str:
        decrypted_content = ''
        for c in encrypted_content:
            decSpecCode = (ord(c) - self.salt - (self.secureKey * self.salt)) % 256
            if decSpecCode < 0:
                decSpecCode += 256
            decrypted_content += chr(decSpecCode)
        return decrypted_content

File: php/NovaCrypt.php
<?php

class NovaCrypt {
    private $secureKey;
    private $salt;

    public function __construct(int $SECURE_KEY, int $SALT) {
        $this->secureKey = $SECURE_KEY;
        $this->salt = ($SALT === 0) ? 9 : $SALT;
    }

    public function encrypt(string $content): string {
        $encryptedContent = '';
        for ($i = 0; $i < strlen($content); $i++) {
            $c = $content[$i];
            $genSpecCode = (ord($c) + ($this->secureKey * $this->salt) + $this->salt) % 256;
            $encryptedContent .= chr($genSpecCode);
        }
        return $encryptedContent;
    }

    public function decrypt(string $encryptedContent): string {
        $decryptedContent = '';
        for ($i = 0; $i < strlen($encryptedContent); $i++) {
            $c = $encryptedContent[$i];
            $decSpecCode = (ord($c) - $this->salt - ($this->secureKey * $this->salt)) % 256;
            if ($decSpecCode < 0) {
                $decSpecCode += 256;
            }
            $decryptedContent .= chr($decSpecCode);
        }
        return $decryptedContent;
    }
}

?>
